unittest/MutiThreadTest.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include <future>
+#include <vector>
+
+#include "knowhere/common/Dataset.h"
+#include "knowhere/index/VecIndex.h"
+
+namespace knowhere {
+
+namespace mthreadtest {
+std::vector<std::future<DatasetPtr>>&
+GetFeatureQueue() {
+    static std::vector<std::future<DatasetPtr>> qu;
+    return qu;
+}
+
+DatasetPtr
+Async(VecIndexPtr& idx, DatasetPtr& dataset, Config& cfg, faiss::BitsetView bitset) {
```

Comment:
I guess we tried to make an API for test team to run code in parallel? If so, suggest to make it as general as possible, which means no any index related thing here. We can try to pass a lambda in.

Reply:
Ok, I will have a try.